packages/react-components/react-tree/src/components/TreeItem/useTreeItem.tsx

```diff
@@ -60,6 +61,50 @@ export function useTreeItem_unstable(props: TreeItemProps, ref: React.Ref<HTMLDi
   const selectionMode = useTreeContext_unstable(ctx => ctx.selectionMode);
   const checked = useTreeContext_unstable(ctx => ctx.checkedItems.get(value) ?? false);
 
+  // This callback ref should be called before the element is removed from the document
+  // If the component is unmounted while it is the active element - do not lose focus
+  const handleFocusedUnmountRef = React.useMemo(() => {
+    let el: HTMLDivElement | null = null;
+
+    return (e: HTMLDivElement | null) => {
+      if (e) {
+        el = e;
```

Comment:
If you use `useIsomorphicLayoutEffect` instead of a memoized ref you can avoid this `el` -> `e` naming problem here and just use `triggerRef` instead